<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1800</width>
    <height>1200</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout"/>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1800</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuExport_to">
     <property name="title">
      <string>Export...</string>
     </property>
     <addaction name="actionSTLExport"/>
    </widget>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="menuExport_to"/>
    <addaction name="actionSave"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="dmlDock">
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dmlWidget"/>
  </widget>
  <widget class="QDockWidget" name="consoleDock">
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="consoleWidget"/>
  </widget>
  <widget class="QDockWidget" name="propDock">
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="propWidget">
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <layout class="QVBoxLayout" name="propLayout">
     <property name="rightMargin">
      <number>9</number>
     </property>
     <property name="bottomMargin">
      <number>9</number>
     </property>
     <item>
      <layout class="QGridLayout" name="geometryBox">
       <item row="1" column="0">
        <widget class="QToolButton" name="diffButton">
         <property name="text">
          <string>Difference</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QToolButton" name="unionButton">
         <property name="text">
          <string>Union</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QToolButton" name="interButton">
         <property name="text">
          <string>Intersection</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QGroupBox" name="simSettingsBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>1</horstretch>
         <verstretch>1</verstretch>
        </sizepolicy>
       </property>
       <property name="maximumSize">
        <size>
         <width>12233333</width>
         <height>400</height>
        </size>
       </property>
       <property name="title">
        <string>Simulation Settings</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QPushButton" name="stepButton">
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>50</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>50</height>
           </size>
          </property>
          <property name="text">
           <string>Step</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="stopButton">
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>50</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>50</height>
           </size>
          </property>
          <property name="text">
           <string>Stop</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1" rowspan="3">
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,1">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>10</number>
            </property>
            <item>
             <widget class="QLabel" name="timestepLabel">
              <property name="text">
               <string>GPU timestep:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="timestep">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximumSize">
               <size>
                <width>75</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="readOnly">
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="1,1">
            <property name="rightMargin">
             <number>10</number>
            </property>
            <item>
             <widget class="QLabel" name="renderUpdateLabel">
              <property name="text">
               <string>Render update: </string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="renderUpdateEdit">
              <property name="maximumSize">
               <size>
                <width>75</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="baseSize">
               <size>
                <width>75</width>
                <height>0</height>
               </size>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="2" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <property name="rightMargin">
             <number>10</number>
            </property>
            <item>
             <widget class="QLabel" name="springConstLabel">
              <property name="text">
               <string>Spring constant:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="springConstEdit">
              <property name="maximumSize">
               <size>
                <width>75</width>
                <height>16777215</height>
               </size>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="3" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <property name="rightMargin">
             <number>10</number>
            </property>
            <item>
             <widget class="QLabel" name="massValueLabel">
              <property name="text">
               <string>Mass value:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="massValueEdit">
              <property name="maximumSize">
               <size>
                <width>75</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="pauseButton">
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>50</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>50</height>
           </size>
          </property>
          <property name="text">
           <string>Pause</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QPushButton" name="startButton">
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>50</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>50</height>
           </size>
          </property>
          <property name="text">
           <string>Start</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLCDNumber" name="timeLCD">
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="palette">
           <palette>
            <active>
             <colorrole role="WindowText">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>0</red>
                <green>0</green>
                <blue>0</blue>
               </color>
              </brush>
             </colorrole>
            </active>
            <inactive>
             <colorrole role="WindowText">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>0</red>
                <green>0</green>
                <blue>0</blue>
               </color>
              </brush>
             </colorrole>
            </inactive>
            <disabled>
             <colorrole role="WindowText">
              <brush brushstyle="SolidPattern">
               <color alpha="96">
                <red>164</red>
                <green>166</green>
                <blue>168</blue>
               </color>
              </brush>
             </colorrole>
            </disabled>
           </palette>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QCheckBox" name="stressCheckBox">
          <property name="text">
           <string> Show stress</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="iconSize">
    <size>
     <width>45</width>
     <height>45</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionwireframeView"/>
   <addaction name="actionSTLView"/>
   <addaction name="actionDMLView"/>
   <addaction name="separator"/>
   <addaction name="actionSimulation_Mode"/>
   <addaction name="separator"/>
   <addaction name="actionRecordSim"/>
   <addaction name="actionSaveSim"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcutVisibleInContextMenu">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionwireframeView">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/resources/icons/wireframe.png</normaloff>:/resources/icons/wireframe.png</iconset>
   </property>
   <property name="text">
    <string>Wireframe View</string>
   </property>
   <property name="toolTip">
    <string>Wireframe View</string>
   </property>
  </action>
  <action name="actionSTLView">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/resources/icons/solid.png</normaloff>:/resources/icons/solid.png</iconset>
   </property>
   <property name="text">
    <string>Model View</string>
   </property>
   <property name="toolTip">
    <string>STL View</string>
   </property>
  </action>
  <action name="actionDMLView">
   <property name="text">
    <string>DML View</string>
   </property>
   <property name="toolTip">
    <string>DML View</string>
   </property>
  </action>
  <action name="actionSimulation_Mode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Simulation Mode</string>
   </property>
   <property name="toolTip">
    <string>Simulation Mode</string>
   </property>
  </action>
  <action name="actionStartSim">
   <property name="text">
    <string>Start</string>
   </property>
   <property name="toolTip">
    <string>Start simulation</string>
   </property>
  </action>
  <action name="actionPauseSim">
   <property name="text">
    <string>Pause</string>
   </property>
   <property name="toolTip">
    <string>Pause simulation</string>
   </property>
  </action>
  <action name="actionStopSim">
   <property name="text">
    <string>Stop</string>
   </property>
   <property name="toolTip">
    <string>Stop simulation</string>
   </property>
  </action>
  <action name="actionStepSim">
   <property name="text">
    <string>Step</string>
   </property>
   <property name="toolTip">
    <string>Step simulation</string>
   </property>
  </action>
  <action name="actionRecordSim">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Record Video</string>
   </property>
   <property name="toolTip">
    <string>Record video</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionSaveSim">
   <property name="text">
    <string>Save Simulation</string>
   </property>
   <property name="toolTip">
    <string>Save simulation</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionSTLExport">
   <property name="text">
    <string>STL</string>
   </property>
   <property name="toolTip">
    <string>STL file export</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionSimulation_Mode</sender>
   <signal>toggled(bool)</signal>
   <receiver>simSettingsBox</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>1645</x>
     <y>610</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stepButton</sender>
   <signal>clicked()</signal>
   <receiver>actionStepSim</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1536</x>
     <y>467</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startButton</sender>
   <signal>clicked()</signal>
   <receiver>actionStartSim</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1536</x>
     <y>635</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pauseButton</sender>
   <signal>clicked()</signal>
   <receiver>actionPauseSim</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1536</x>
     <y>579</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stopButton</sender>
   <signal>clicked()</signal>
   <receiver>actionStopSim</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1536</x>
     <y>523</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
